packages/bee-q/src/components/option/bq-option.tsx

```diff
@@ -0,0 +1,178 @@
+import { h, Component, Prop, Element, State, Event, EventEmitter, Listen, Method } from '@stencil/core';
+
+import { hasSlotContent } from '../../shared/utils';
+
+/**
+ * @part label - The `span` element in which the label text is displayed
+ * @part prefix - The `span` element in which the prefix is displayed (generally `bq-icon`)
+ * @part suffix - The `span` element in which the suffix is displayed (generally `bq-icon`)
+ */
+@Component({
+  tag: 'bq-option',
+  styleUrl: './scss/bq-option.scss',
+  shadow: true,
+})
+export class BqOption {
+  // Own Properties
+  // ====================
+
+  private prefixElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqOptionElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() hasPrefix: boolean = false;
+
+  /** If true, option element is part of a group */
+  @State() isOptionInGroup: boolean = false;
```

Comment:
I would replace this `State` with a property `inset` / `indent` defaulted to `false`.